src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationContext.java

```diff
@@ -94,6 +94,16 @@ public final class CcCompilationContext implements CcCompilationContextApi<Artif
   // This is needed only when code coverage collection is enabled, to report the actual source file
   // name in the coverage output file.
   private final NestedSet<Tuple> virtualToOriginalHeaders;
+
+  /**
+   * Preserves mappings of _virtual_include directories to original paths relative to
+   * the workspace directory.  Intended for use by compiler options like GCC's
+   * {@code -fdebug-prefix-map}.  See {@link virtualToOriginalHeaders} for additional
+   * context.  Unlike {@link virtualToOriginalHeaders}, this is populated even if
+   * coverage isn't enabled.
+   */
+  private final NestedSet<Tuple> virtualToOriginalDirs;
```

Comment:
This new field will potentially result in memory regression. There's already `virtualToOriginalHeaders`, which is probably even larger, containing all header files, however it's only populated with coverage enabled.

I wonder if there's a more efficient way to reconstruct the information from `_virtual_includes` constant. It would be quite hacky though, but a wrapper script should be able to do it.